test/models/project_test.rb

```diff
@@ -0,0 +1,71 @@
+require 'test_helper'
+
+class ProjectTest < ActiveSupport::TestCase
+  attr_reader :project
+
+  test "it is valid" do
+    project = build(:project)
+    project.categories << create(:category)
+    assert project.valid?
+  end
+
+  test "it cannot have an empty string as a title" do
+    category = Category.create(name: "hot beverages")
+    project = category.projects.create(title: '', description: "this is black gold", price: 30000)
+    assert project.invalid?
+  end
+
+  test "it cannot have an empty string as a description" do
+    category = Category.create(name: "hot beverages")
+
+    project = category.projects.create(title: 'espresso', description: "", price: 30000)
```

Comment:
Line is too long. [88/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.